javav2/example_code/ec2/src/main/java/com/example/ec2/scenario/EC2Actions.java

```diff
@@ -0,0 +1,704 @@
+    }
+    // snippet-end:[ec2.java2.scenario.disassociate_address.main]
+
+    // snippet-start:[ec2.java2.associate_address.main]
+    /**
+     * Associates an Elastic IP address with an EC2 instance asynchronously.
+     *
+     * @param instanceId    the ID of the EC2 instance to associate the Elastic IP address with
+     * @param allocationId  the allocation ID of the Elastic IP address to associate
+     * @return a {@link CompletableFuture} that completes with the association ID when the operation is successful,
+     *         or throws a {@link RuntimeException} if the operation fails
+     */
+    public CompletableFuture<String> associateAddressAsync(String instanceId, String allocationId) {
+        AssociateAddressRequest associateRequest = AssociateAddressRequest.builder()
+            .instanceId(instanceId)
+            .allocationId(allocationId)
+            .build();
+
+        CompletableFuture<AssociateAddressResponse> responseFuture = getAsyncClient().associateAddress(associateRequest);
+        return responseFuture.thenApply(response -> {
+            if (response.associationId() != null) {
+                return response.associationId();
+            } else {
+                throw new RuntimeException("Association ID is null after associating address.");
+            }
+        }).whenComplete((result, ex) -> {
+            if (ex != null) {
+                throw new RuntimeException("Failed to associate address", ex);
+            }
+        });
+    }
+    // snippet-end:[ec2.java2.associate_address.main]
+
+    // snippet-start:[ec2.java2.allocate_address.main]
+    /**
+     * Allocates an Elastic IP address asynchronously in the VPC domain.
+     *
+     * @return a {@link CompletableFuture} containing the allocation ID of the allocated Elastic IP address
+     */
+    public CompletableFuture<String> allocateAddressAsync() {
+        AllocateAddressRequest allocateRequest = AllocateAddressRequest.builder()
+            .domain(DomainType.VPC)
+            .build();
+
+        CompletableFuture<AllocateAddressResponse> responseFuture = getAsyncClient().allocateAddress(allocateRequest);
+        return responseFuture.thenApply(AllocateAddressResponse::allocationId).whenComplete((result, ex) -> {
+            if (ex != null) {
+                throw new RuntimeException("Failed to allocate address", ex);
+            }
+        });
+    }
+    // snippet-end:[ec2.java2.allocate_address.main]
+
+    // snippet-start:[ec2.java2.scenario.describe_instance.main]
+    /**
+     * Asynchronously describes the state of an EC2 instance.
+     *
+     * @param newInstanceId the ID of the EC2 instance to describe
+     * @return a {@link CompletableFuture} that, when completed, contains a string describing the state of the EC2 instance
+     */
+    public CompletableFuture<String> describeEC2InstancesAsync(String newInstanceId) {
+        DescribeInstancesRequest request = DescribeInstancesRequest.builder()
```

Comment:
This should be a paginator.

https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/ec2/Ec2Client.html#describeInstancesPaginator()